compiler/luci/pass/src/CircleQuantizer.cpp

```diff
@@ -69,12 +70,15 @@ class QuantizeOptionsImpl final : public luci::CircleQuantizer::Options
   const std::string param(AlgorithmParameters) const final;
   void params(AlgorithmParameters, std::vector<std::string> &) final;
   std::vector<std::string> params(AlgorithmParameters) const final;
+  void layer_params(AlgorithmParameters, std::vector<std::shared_ptr<LayerParam>> &) final;
+  std::vector<std::shared_ptr<LayerParam>> layer_params(AlgorithmParameters) const final;
   bool query(Algorithm) final;
 
 private:
   std::vector<Algorithm> _algorithms;
   std::map<AlgorithmParameters, const std::string> _algorithm_params;
   std::map<AlgorithmParameters, std::vector<std::string>> _multiple_params;
+  std::map<AlgorithmParameters, std::vector<std::shared_ptr<LayerParam>>> _multiple_layer_params;
```

Comment:
is this need to be `shared_ptr` ?

Reply:
I think there can be two other options.

1. Use `LayerParam`
  - This will cause copy whenever we access the parameter by `layer_params`. Copy overhead would be acceptable for now, but I just chose to avoid copy because LayerParam can be extended in the future.

2. Use `LayerParam*`
  - I didn't choose this option, because I don't want to track the lifetime of LayerParam.

Reply:
Sorry for not asking what I wanted to ask. ONE usually use `unique_ptr`. this uses `shared_ptr`. reason is `unique_ptr` is lighter than `shared_ptr`. so I asked why use `shared_ptr`.

Reply:
Ah, I misunderstood your comment :)

> that is, is this shared among other objects that own this?

It is not shared for now, but I'm not sure it will be in the future. That is the reason I used `shared_ptr` instead of `unique_ptr`. Performance difference would be negligible.

Reply:
OK :)